pywinauto/base_application.py

```diff
@@ -407,149 +557,41 @@ def exists(self, timeout=None, retry_interval=None):
-
-             * 'exists' means that the window is a valid handle
-             * 'visible' means that the window is not hidden
-             * 'enabled' means that the window is not disabled
-             * 'ready' means that the window is visible and enabled
-             * 'active' means that the window is active
-
-        :param timeout: Raise an :func:`pywinauto.timings.TimeoutError` if the window
-            is not in the appropriate state after this number of seconds.
-            Default: :py:attr:`pywinauto.timings.Timings.window_find_timeout`.
-
-        :param retry_interval: How long to sleep between each retry.
-            Default: :py:attr:`pywinauto.timings.Timings.window_find_retry`.
-
-        An example to wait until the dialog
-        exists, is ready, enabled and visible: ::
+        ctrl = self.find()
 
-            self.Dlg.wait("exists enabled visible ready")
+        if ctrl.is_dialog():
+            # dialog controls are all the control on the dialog
+            dialog_controls = ctrl.children()
 
-        .. seealso::
-            :func:`WindowSpecification.wait_not()`
-
-            :func:`pywinauto.timings.TimeoutError`
-        """
-        check_method_names, timeout, retry_interval = self.__parse_wait_args(wait_for, timeout, retry_interval)
-        wait_until(timeout, retry_interval,
-                   lambda: self.__check_all_conditions(check_method_names, retry_interval))
-
-        # Return the wrapped control
-        return self.find()
-
-    def wait_not(self, wait_for_not, timeout=None, retry_interval=None):
-        """
-        Wait for the window to not be in a particular state/states.
-
-        :param wait_for_not: The state to wait for the window to not be in. It can be any
-            of the following states, also you may combine the states by space key.
-
-             * 'exists' means that the window is a valid handle
-             * 'visible' means that the window is not hidden
-             * 'enabled' means that the window is not disabled
-             * 'ready' means that the window is visible and enabled
-             * 'active' means that the window is active
-
-        :param timeout: Raise an :func:`pywinauto.timings.TimeoutError` if the window is sill in the
-            state after this number of seconds.
-            Default: :py:attr:`pywinauto.timings.Timings.window_find_timeout`.
-
-        :param retry_interval: How long to sleep between each retry.
-            Default: :py:attr:`pywinauto.timings.Timings.window_find_retry`.
-
-        An example to wait until the dialog is not ready, enabled or visible: ::
+            ctrls_to_print = dialog_controls[:]
+            # filter out hidden controls
+            ctrls_to_print = [
+                ctrl for ctrl in ctrls_to_print if ctrl.is_visible()]
+        else:
+            dialog_controls = ctrl.top_level_parent().children()
+            ctrls_to_print = [ctrl]
```

Comment:
Var `ctrls_to_print` is never used. Is it some mistake or typo in variable name?